sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,326 @@
+
+  /**
+   * Defines the expression that the rule evaluates. Only the messages whose CorrelationId match the CorrelationId set in the filter expression are allowed. Only one between a correlation and a sql expression can be defined.
+   */
+  correlationFilter?: CorrelationFilterOptions;
+
+  /**
+   * Defines the expression that the rule evaluates as a true filter.
+   */
+  trueFilter?: string;
+
+  /**
+   * Defines the expression that the rule evaluates as a false filter.
+   */
+  falseFilter?: string;
+
+  /**
+   * Defines the expression that the rule evaluates. If the rule is of type SQL, the expression string is interpreted as a SQL92 expression which must evaluate to True or False. If the rule is of type CorrelationFilterExpression then only the messages whose CorrelationId match the CorrelationId set in the filter expression are allowed.
+   */
+  sqlRuleAction?: string;
+}
+
+/**
+ * Represents all attributes of a rule entity
+ */
+export interface Rule {
+  /**
+   * Name of the rule
+   */
+  ruleName?: string;
+
+  /**
+   * Name of topic
+   */
+  topicName?: string;
+
+  /**
+   * Name of subscription
+   */
+  subscriptionName?: string;
+
+  /**
+   * Represents Topic filter.
+   */
+  filter?: Filter;
+
+  /**
+   * Rule action
+   */
+  action?: string;
+
+  /**
+   * Created at timestamp
+   */
+  createdAt?: string;
+}
+
+/**
+ * Represents all possible fields on Topic Filter
+ */
+export interface Filter extends CorrelationFilterOptions {
+  /**
+   * SQL expression to use.
+   */
+  sqlExpression?: string;
+
+  /**
+   * CompatibilityLevel field on the SQL rule
+   */
+  compatibilityLevel: number;
+}
+
+/**
+ * Represents settable options on CorrelationFilter
+ */
+export interface CorrelationFilterOptions {
+  /**
+   * CorrelationId value
+   */
+  correlationId?: string;
+
+  /**
+   * Message label
+   */
+  label?: string;
+
+  /**
+   * Message ID
+   */
+  messageId?: number;
+
+  /**
+   * Message 'To' field
+   */
+  to?: number;
+
+  /**
+   * Message 'ReplyTo' field
+   */
+  replyTo?: number;
+
+  /**
+   * Message 'SessionId' field
+   */
+  sessionId?: string;
+
+  /**
+   * Message 'ReplyToSessionId' field
+   */
+  replyToSessionId?: string;
+
+  /**
+   * Message 'ContentType' field
+   */
+  contentType?: string;
+
+  /**
+   * Custom key value properties.
+   */
+  properties?: any;
+}
+
+/**
+ * @ignore RuleResourceSerializer for serializing / deserializing Rule entities
+ */
+export class RuleResourceSerializer implements AtomXmlSerializer {
+  serialize(rule: InternalRuleOptions): string {
+    const resource: { Name: any; Filter: any; Action: any } = {
+      Name: rule.name,
+      Filter: [],
+      Action: []
+    };
+
+    if (rule) {
+      const filters = [];
+      if (rule.sqlExpressionFilter) {
+        const sqlFilter = {
+          $: {
+            type: "SqlFilter"
+          },
+          SqlExpression: rule.sqlExpressionFilter,
+          CompatibilityLevel: 20
```

Comment:
How does this compatibility level get specified/what does it mean? Should it be a constant and/or configurable?

Reply:
It's being treated as a constant in the older SDK. At this time, create/update on rules is having issues - this will need to be looked into later. Filed issue for it at - https://github.com/Azure/azure-sdk-for-js/issues/5339

Reply:
According to the service's public documentation, this was exposed for convenience but currently takes a fixed default of 20. Current changes and documentation on it reflects the same.